### ISSUE TITLE
Regression in refreshing modules with signal in 0.9.3

### ISSUE THREAD

**Issue by maximbaz** [Alexays/Waybar#803]
Add this to your config:

```
    "custom/date": {
        "exec": "printf '{\"text\": \"%s\"}\n' \"$(date +%s)\"",
        "return-type": "json",
        "signal": 10,
        "on-click": "pkill -RTMIN+10 -x waybar",
        "interval": 1
    },
```

You will see timestamp running. Now click on the module.

Expected: it will refresh
Actual: the module will disappear forever

Worked fine in 0.9.2

I couldn't bisect to a specific commit because some commits fail to compile :(

**Comment by Alexays** [comment]
Should be fixed with #778
Can you test and confirm?

**Comment by maximbaz** [comment]
Tested and confirm, thanks. Should have searched better, will close this as duplicate of #778 and #739.
